src/test/java/de/tum/in/www1/artemis/service/GitlabCIServiceTest.java

```diff
@@ -0,0 +1,138 @@
+package de.tum.in.www1.artemis.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+
+import org.gitlab4j.api.GitLabApiException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationGitlabCIGitlabSamlTest;
+import de.tum.in.www1.artemis.ContinuousIntegrationTestService;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.exception.GitLabCIException;
+
+public class GitlabCIServiceTest extends AbstractSpringIntegrationGitlabCIGitlabSamlTest {
```

Comment:
The `public` is not necessary for test classes and methods in JUnit 5 any more. They can just be package-private.